src/components/topo/ucc_topo.h

```diff
@@ -126,6 +134,58 @@ static inline int ucc_topo_isoppn(ucc_topo_t *topo)
     return ucc_topo_max_ppn(topo) == ucc_topo_min_ppn(topo);
 }
 
+/* Returns min socket size across the nodes */
+static inline ucc_rank_t ucc_topo_min_socket_size(ucc_topo_t *topo)
+{
+    ucc_sbgp_t *sbgp = ucc_topo_get_sbgp(topo, UCC_SBGP_NODE_LEADERS);
+
+    if (sbgp->status == UCC_SBGP_NOT_INIT) {
+        ucc_assert(topo->topo->sock_bound);
```

Comment:
why is this assertion here? what if node leaders sbgp is successfully created but proc is not sock/numa bound ? will this functino return topo->min_socket_size = 0? should the caller of the fn treat return 0 as non correct value?

Reply:
This function - ucc_topo_min_socket_size() - should only be called if sock bound. there are similar functions for numa as well.
If user calls ucc_topo_get_sbgp(topo, UCC_SBGP_NODE_LEADERS) outside of this function then he wont reach this assertion.
In any case  topo->min_socket_size is set by default to UCC_MAX_RADIX (not to 0), if there is an error along the way it is set to UCC_RANK_INVALID. User should check. In our case we take in team ucc_min() of the result and of team size, team size will always be smaller then UCC_MAX_RADIX/UCC_RANK_INVALID so there won't be an issue and no need to directly check.

Reply:
i see. i'd say it is convoluted and a not obvious. lets at least document this in a comment. also, if we say that user always has to call this fn with sock/numa bound then assertion should be moved outside of "if (stats == NOT_INIT)" to the top of the fn. correct?